Source/NetworkClients/ClientProtocol.swift

```diff
@@ -0,0 +1,65 @@
+//  Copyright © 2016 Gavan Chan. All rights reserved.
+
+import Argo
+import Foundation
+import RxCocoa
+import RxSwift
+
+fileprivate let apiKey: String = "402c2de16506bb59c7a6afc8b60778c2"
+
+extension URLRequest {
+  typealias Response = (HTTPURLResponse, Data)
+
+  enum Method {
+    case get
+  }
+}
+
+protocol ClientProtocol {
+  associatedtype Endpoint
+
+  var base: URL { get }
+  var session: URLSession { get }
+
+  func request(
+    for endpoint: Endpoint,
+    via method: URLRequest.Method,
+    withQuery query: Dictionary<String, String>) -> URLRequest
+}
+
+extension ClientProtocol where Endpoint: RawRepresentable, Endpoint.RawValue == String {
```

Comment:
While passable, this constrained protocol extension makes the assumption that the `rawValue` of the `RawRepresentable` `Endpoint` type represents the endpoint's path.

This is probably better extracted out to some other matching protocol declaration for readability, and this protocol extension changed to match.

Let's raise an issue for consideration going forward.